download/releaseNotes/releaseNotes7.adoc

```diff
@@ -5,6 +5,20 @@
 :awestruct-release_notes_version: 7
 :awestruct-release_notes_version_qualifier: Final
 
+== New and noteworthy: Engine 7.32.0.Final
+
+=== `groupBy()` support for constraint streams of all cardinalities
+
+We continue improving the constraint streams API.
+In this release, we have added capabilities to modify your streams using the https://docs.optaplanner.org/7.31.0.Final/optaplanner-docs/html_single/index.html#constraintStreamsGroupingAndCollectors[`groupBy()` building block].
+
+== New and noteworthy: Engine 7.31.0.Final
+
+=== QuadStreams
+
+The constraint streams API received a major update, introducing support for streams of 4 variables.
+Please refer to https://docs.optaplanner.org/7.31.0.Final/optaplanner-docs/html_single/index.html#constraintStreamsCardinality[constraint streams documentation] for details.
```

Comment:
Never link an actual version number. Use the latestFinal symbolic link instead: https://docs.optaplanner.org/latestFinal/optaplanner-docs/html_single/ 
Why? If someone is currently using 7.12 and wants to upgrade to 7.35 later this year, they might still read this n&n section and if they click the link, we want them to look at the latest and greatest.